components/permalink_view/actions.ts

```diff
@@ -53,13 +54,16 @@ export function focusPost(postId, returnTo = '', currentUserId) {
                 browserHistory.replace(`/error?type=${ErrorPageTypes.PERMALINK_NOT_FOUND}&returnTo=${returnTo}`);
                 return;
             }
+            let membership;
+            if (typeof currentUserId !== 'undefined') {
+                membership = await dispatch(getChannelMember(channel.id, currentUserId));
+            }
 
-            const membership = await dispatch(getChannelMember(channel.id, currentUserId));
             if ('data' in membership) {
                 myMember = membership.data;
             }
 
-            if (!myMember) {
+            if (!myMember && typeof currentUserId !== 'undefined') {
```

Comment:
```suggestion
            if (!myMember && currentUserId) {
```
if condition will automatically check for `undefined`